core/src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -189,6 +189,13 @@ public DiscoveryNode localNode() {
     }
 
     @Override
+    public List<String> localCustomMetaDataTypes() {
+        List<String> customMetaDataList = new ArrayList<>(MetaData.customPrototypes.keySet());
```

Comment:
I think relying on order of elements in the list is brittle. I think representing this list as a set might be a more reliable solution.